frontend/views/containers/chatroom/SendArea.vue

```diff
@@ -390,6 +435,64 @@ export default ({
         bottom: `${innerHeight - bbox.top + 8}px` // 8 -> 0.5rem gap
       })
     },
+    openFileAttach (e) {
+      e.target.blur()
+      this.$refs.fileAttachmentInputEl.click()
+    },
+    fileAttachmentHandler (filesList, appendItems = false) {
+      const getFileExtension = name => {
+        const lastDotIndex = name.lastIndexOf('.')
+        return lastDotIndex === -1 ? '' : name.substring(lastDotIndex)
+      }
+      const attachmentsExist = Boolean(this.ephemeral.attachment.length)
+      const list = appendItems && attachmentsExist
+        ? [...this.ephemeral.attachment]
+        : []
+
+      if (attachmentsExist) {
+        // make sure to clear the previous state if there is already attached file(s).
+        this.clearAllAttachments()
+      }
+
+      for (const file of filesList) {
+        const fileExt = getFileExtension(file.name)
+        const fileUrl = URL.createObjectURL(file)
+        const fileSize = file.size
+
+        if (fileSize > Math.pow(10, 9)) {
+          // TODO: update Math.pow(10, 9) above with the value delivered from the server once it's implemented there.
+          return sbp('okTurtles.events/emit', OPEN_MODAL, 'ChatFileAttachmentWarningModal', { type: 'large' })
+        } else if (!fileExt || !CHAT_ATTACHMENT_SUPPORTED_EXTENSIONS.includes(fileExt)) {
+          // Give users a warning about unsupported file types
+          return sbp('okTurtles.events/emit', OPEN_MODAL, 'ChatFileAttachmentWarningModal', { type: 'unsupported' })
```

Comment:
The file extension needs to be lowercased before it is compared. This way files like `file.JPG` are accepted too.